### PR TITLE
Disable official buttons option in Sharing Buttons block

### LINKED ISSUE #34580: Sharing Block: Disable `official buttons` setting in Sharing Buttons block
Currently, we are not supporting `official buttons` view in Sharing Buttons block so we wan't to disable this option for now to avoid confusion.

----

Epic: #19694

### PR DESCRIPTION
**Pull request by andrii-lysenko**
Fixes https://github.com/Automattic/jetpack/issues/34580

## Proposed changes:

As this feature is not available yet, we want to mark this option disabled.
I've also added localization to the options.

Also, open to change it so we just remove the button entirely as mentioned https://github.com/Automattic/jetpack/issues/34580#issuecomment-1852331255, as don't have a strong opinion on this

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

- Spin up your site with local changes
- Make sure you have beta block enabled. In your environment (example Docker) wp-config.php you should have define( 'JETPACK_BLOCKS_VARIATION', 'beta' );
- Add Sharing Buttons block.
- In settings (right block sidebar), `Official buttons` should not be present

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/sharing-buttons/edit.js [hunk 1]
```diff
@@ -32,10 +32,14 @@ export function SharingButtonsEdit( props ) {
...
-							{ value: 'text', label: 'Text Only' },
-							{ value: 'official', label: 'Official Buttons' },
+							/* translators: Sharing: Sharing button option label. */
+							{ value: 'icon-text', label: __( 'Icon & Text', 'jetpack' ) },
+							/* translators: Sharing: Sharing button option label. */
+							{ value: 'icon', label: __( 'Icon Only', 'jetpack' ) },
+							/* translators: Sharing: Sharing button option label. */
+							{ value: 'text', label: __( 'Text Only', 'jetpack' ) },
+							/* translators: Sharing: Sharing button option label. */
+							{ value: 'official', label: __( 'Official Buttons', 'jetpack' ), disabled: true },
```
Comment (andrii-lysenko) [inline projects/plugins/jetpack/extensions/blocks/sharing-buttons/edit.js]: I could probably use wording like 'Official Buttons (In Development)' or something. Or leave as is.

**Comment by jeherve** [comment]
> Also, open to change it so we just remove the button entirely

That would be my vote as well, I think it avoids setting expectations too early.

**Comment by andrii-lysenko** [comment]
@jeherve Thanks! Removed the button.

**Review by monsieur-z — APPROVED** [review]
Works as expected 🚢 

The comments for translators might not provide enough context if you ask me.

**Comment by andrii-lysenko** [comment]
@monsieur-z Thanks! I used them from Calypso, so my assumption is that it's already has a translation, so I just wanted to keep things in line.

**Pull request merged**